src/mapgen_v7.cpp

```diff
@@ -513,8 +519,11 @@ int MapgenV7::generateBaseTerrain()
 		heightmap[index]       = surface_y;
 		ridge_heightmap[index] = surface_y;
 
-		if (surface_y > stone_surface_max_y)
-			stone_surface_max_y = surface_y;
+		if (surface_y > *stone_surface_max_y)
```

Comment:
Hmm, maybe it'd be a good idea to keep these parameters in temporary local variables, and only set them right before returning.  This offers better performance because the compiler is certain that nothing may alias these pointers.